tools/accuracy_checker/custom_evaluators/colorization_evaluator.py

```diff
@@ -0,0 +1,220 @@
+            progress_reporter.reset(self.dataset.size)
+
+        for batch_id, (dataset_indices, batch_annotation) in enumerate(self.dataset):
+            batch_identifiers = [annotation.identifier for annotation in batch_annotation]
+            batch_input = [self.reader(identifier=identifier) for identifier in batch_identifiers]
+            batch_input = self.preprocessing_executor.process(batch_input, batch_annotation)
+            batch_input, _ = extract_image_representations(batch_input)
+            batch_out = np.array(self.test_model.predict(batch_annotation, batch_input))
+            batch_prediction = self.check_model.predict(batch_identifiers, batch_out)
+            self.metric_executor.update_metrics_on_batch(dataset_indices, batch_annotation, batch_prediction)
+            if self.metric_executor.need_store_predictions:
+                self._annotations.extend(batch_annotation)
+                self._predictions.extend(batch_prediction)
+            progress_reporter.update(batch_id, len(batch_prediction))
+
+        if progress_reporter:
+            progress_reporter.finish()
+
+    def compute_metrics(self, print_results=True, ignore_results_formatting=False):
+        if self._metrics_results:
+            del self._metrics_results
+            self._metrics_results = []
+
+        for result_presenter, evaluated_metric in self.metric_executor.iterate_metrics(
+            self._annotations, self._predictions):
+            self._metrics_results.append(evaluated_metric)
+            if print_results:
+                result_presenter.write_result(evaluated_metric, ignore_results_formatting)
+
+        return self._metrics_results
+
+    def print_metrics_results(self, ignore_results_formatting=False):
+        if not self._metrics_results:
+            self.compute_metrics(True, ignore_results_formatting)
+            return
+        result_presenters = self.metric_executor.get_metric_presenters()
+        for presenter, metric_result in zip(result_presenters, self._metrics_results):
+            presenter.write_results(metric_result, ignore_results_formatting)
+
+    def release(self):
+        self.launcher.release()
+
+    def reset(self):
+        self.metric_executor.reset()
+
+    @staticmethod
+    def get_processing_info(config):
+        module_specific_params = config.get('module_config')
+        model_name = config['name']
+        dataset_config = module_specific_params['datasets'][0]
+        launcher_config = module_specific_params['launchers'][0]
+        return (
+            model_name, launcher_config['framework'], launcher_config['device'], launcher_config.get('tags'),
+            dataset_config['name']
+        )
+
+
+class BaseModel:
+    def __init__(self, network_info, launcher):
+        self.network_info = network_info
+        self.supported_format = ['.xml', '.bin']
+
+    def check_format(self, current_format):
+        if not os.path.splitext(current_format)[1] in self.supported_format:
```

Comment:
we avoid to use `os.path` as very old and low level lib, please use `pathlib.Path`
BTW,  Path object usage allow avoid casting to string and make check more elegant (for info see https://docs.python.org/3/library/pathlib.html#pathlib.PurePath.suffix):
```python
if not current_path.suffix in self.supported_format:
```

Reply:
@eaidova corrected)